docs/ERC721.md

```diff
@@ -434,7 +434,7 @@ None.
 
 Transfers `tokenId` token from `from_` to `to`.
 
-> Note that this function should be used instead of `safeTransferFrom` to transfer tokens. Exercise caution as tokens sent to a contract that does not support ERC721 can be lost forever.
+> Usage of this method is discouraged, use [`safeTransferFrom`](#safetransferfrom) whenever possible.
```

Comment:
Why is it discouraged? They're used in different scenarios and some people even prefer this method to avoid calling arbitrary contracts and thus prevent side effects

Reply:
https://www.paradigm.xyz/2021/08/the-dangers-of-surprising-code

Reply:
Great point! And thank you for the article^

Reply:
Should we mention something here?
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC721/IERC721.sol#L84

Reply:
just asked Fran, he confirmed it's an old comment and he's not sure about that recommendation. let's not discourage it.

Reply:
https://github.com/OpenZeppelin/openzeppelin-contracts/issues/3477